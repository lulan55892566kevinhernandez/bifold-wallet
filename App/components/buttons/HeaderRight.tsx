import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

import Arrow from '../../assets/icons/large-arrow.svg'
import { ColorPallet, TextTheme } from '../../theme'

interface HeaderButtonProps {
  title: string
  testID?: string
  accessibilityLabel?: string
  onPress?: () => void
  disabled?: boolean
}

const style = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  touchableArea: {
    marginRight: 14,
  },
  title: {
    ...TextTheme.label,
    color: ColorPallet.grayscale.white,
    marginRight: 4,
  },
  icon: {
    transform: [{ rotate: '180deg' }],
  },
})

const HeaderRight: React.FC<HeaderButtonProps> = ({ title, testID, accessibilityLabel, onPress, disabled = false }) => {
  const accessible = accessibilityLabel && accessibilityLabel !== '' ? true : false

  return (
    <TouchableOpacity
      accessibilityLabel={accessibilityLabel}
      accessible={accessible}
      testID={testID}
      onPress={onPress}
      style={[style.touchableArea]}
      disabled={disabled}
    >
      <View style={style.container}>
        <Text style={[style.title]}>{title}</Text>
        <Arrow
          height={TextTheme.label.fontSize}
          width={TextTheme.label.fontSize}
          fill={ColorPallet.grayscale.white}
          style={[style.icon]}
        />
      </View>
    </TouchableOpacity>
  )
}

export default HeaderRight
