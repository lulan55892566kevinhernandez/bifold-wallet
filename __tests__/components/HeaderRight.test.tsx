import { render } from '@testing-library/react-native'
import React from 'react'

import HeaderRight from '../../App/components/buttons/HeaderRight'

describe('Header Right Button Component', () => {
  test('Renders correctly', () => {
    const tree = render(
      <HeaderRight
        title={'Hello Header'}
        accessibilityLabel={'Hello'}
        onPress={() => {
          return
        }}
      />
    )

    expect(tree).toMatchSnapshot()
  })
})
