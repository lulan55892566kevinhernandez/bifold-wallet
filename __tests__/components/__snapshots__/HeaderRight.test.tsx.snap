// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Right Button Component Renders correctly 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 14,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
          "marginRight": 4,
        }
      }
    >
      Hello Header
    </Text>
    <
      fill="#FFFFFF"
      height={15}
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": "180deg",
            },
          ],
        }
      }
      width={15}
    />
  </View>
</View>
`;
